### PR TITLE
add risc-v toolchain

### PR DESCRIPTION
**Pull request by kaspar030**
This PR adds the RISC-V toolchain.

### CODE REVIEW

**Comment by kaspar030** [comment]
Got a problem:

```
riscv-none-embed-ar: error while loading shared libraries: libfl.so.2: cannot open shared object file: No such file or directory
```
... but ubuntu xenial's libfl is some black magic hack to libfl_pic.a ?

**Comment by kaspar030** [comment]
> Got a problem:

Fixed it through injecting Arch Linux' libfl.so.2.0.0 ...

File: Dockerfile [hunk 1]
```diff
@@ -103,6 +103,18 @@ RUN mkdir -p /opt && \
 ENV PATH $PATH:/opt/mips-mti-elf/2016.05-03/bin
 ENV MIPS_ELF_ROOT /opt/mips-mti-elf/2016.05-03
 
+# Install RISC-V binary toolchain
+RUN mkdir -p /opt && \
+        wget -q https://github.com/gnu-mcu-eclipse/riscv-none-gcc/releases/download/v7.2.0-2-20180110/gnu-mcu-eclipse-riscv-none-gcc-7.2.0-2-20180111-2230-centos64.tgz -O- \
+        | tar -C /opt -xz
+
+# HACK download arch linux' flex dynamic library
```
Comment (kaspar030) [inline Dockerfile]: I know this is ugly... Hopefully bionic beaver will solve the issue.
Or are there other options?

Comment (gebart) [reply Dockerfile]: Is there no RISC-V toolchain for Ubuntu?

Comment (kaspar030) [reply Dockerfile]: You mean, pre-packaged? Didn't find one.
The eclipse one used in this PR claims "tested on Debian, Ubuntu, ...". :/

**Comment by kaspar030** [comment]
This is blocking RISC-V-support. I propose going with the hack for now and revisiting when migrating to bionic beaver.

**Comment by gebart** [comment]
Right, I don't mind

**Comment by kaspar030** [comment]
Let's go.

**Comment by kaspar030** [comment]
- squashed, waiting for travis

**Comment by kaspar030** [comment]
&go

**Pull request merged**